From 323308f6d2cd418042467246528f07587fbc151d Mon Sep 17 00:00:00 2001
From: David Pacheco <dap@joyent.com>
Date: Fri, 7 Apr 2017 13:38:19 -0700
Subject: [PATCH] MANTA-3111 marlin agent cueball reports "read ECONNRESET" and
 408 errors Reviewed by: Alex Wilson <alex.wilson@joyent.com> Approved by:
 Alex Wilson <alex.wilson@joyent.com>

---
 agent/lib/agent/agent.js | 11 +++++++++++
 dev/package.json         |  2 +-
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/agent/lib/agent/agent.js b/agent/lib/agent/agent.js
index c4a37ab..2a89209 100644
--- a/agent/lib/agent/agent.js
+++ b/agent/lib/agent/agent.js
@@ -257,6 +257,8 @@ var maRequestTimeout 		= 300000;	/* local long poll timeout */
 var maUpstreamSpareConnsDefault		= 16;
 var maUpstreamMaxConnsDefault		= 512;
 var maUpstreamTcpKeepAliveDelayDefault	= 15000;
+var maUpstreamPingPath			= '/';
+var maUpstreamPingInterval		= 60000;
 
 var maUpstreamSrvMaxTimeout		= 5000;
 
@@ -464,11 +466,17 @@ function mAgent(filename)
 	    maUpstreamSpareConnsDefault;
 	this.ma_http_keepalive_delay = tunables['httpTcpKeepAliveDelay'] ||
 	    maUpstreamTcpKeepAliveDelayDefault;
+	this.ma_http_ping_path = tunables['httpPingPath'] ||
+	    maUpstreamPingPath;
+	this.ma_http_ping_interval = tunables['httpPingInterval'] ||
+	    maUpstreamPingInterval;
 	this.ma_log.info({
 	    'config': this.ma_conf,
 	    'http_nmaxconns': this.ma_http_nmaxconns,
 	    'http_nspares': this.ma_http_nspares,
 	    'http_keepalive_delay': this.ma_http_keepalive_delay,
+	    'http_ping_path': this.ma_http_ping_path,
+	    'http_ping_interval': this.ma_http_ping_interval,
 	    'cueball_recovery': this.ma_cueball_recovery
 	}, 'configuration');
 
@@ -518,6 +526,9 @@ function mAgent(filename)
 	    'log': this.ma_log.child({ 'component': 'CueballAgent' }),
 	    'spares': this.ma_http_nspares,
 	    'maximum': this.ma_http_nmaxconns,
+	    'ping': this.ma_http_ping_path,
+	    'pingInterval': this.ma_http_ping_path ?
+	        this.ma_http_ping_interval : undefined,
 	    'initialDomains': [ this.ma_manta_host ],
 	    'tcpKeepAliveInitialDelay': this.ma_http_keepalive_delay
 	});
diff --git a/dev/package.json b/dev/package.json
index 3368300..46d11e4 100644
--- a/dev/package.json
+++ b/dev/package.json
@@ -16,7 +16,7 @@
 		"assert-plus": "^1.0.0",
 		"bunyan": "0.20.0",
 		"catstreams": "0.5.0",
-		"cueball": "^1.1.4",
+		"cueball": "2.2.4",
 		"dtrace-provider": "0.6.0",
 		"extsprintf": "^1.3.0",
 		"fuzzstream": "0.0.1",
-- 
2.21.0

